sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -661,11 +661,6 @@ def test_check_input_false():
     clf = ElasticNet(selection='cyclic', tol=1e-8)
     # Check that no error is raised if data is provided in the right format
     clf.fit(X, y, check_input=False)
-    X = check_array(X, order='F', dtype='float32')
-    clf.fit(X, y, check_input=True)
-    # Check that an error is raised if data is provided in the wrong dtype,
-    # because of check bypassing
-    assert_raises(ValueError, clf.fit, X, y, check_input=False)
```

Comment:
I would suggest to change this to a smoke test:

```
clf.fit(X, y, check_input=False)
```

and add a comment saying because `check_input=False`, an exhaustive check is not made on `y` but just the dtype of `y` is cast in `_preprocess_data` to the dtype of X so this passes. (We will definitely forget in the future)